### PR TITLE
Add option -include-footer.

### PR DESCRIPTION
**Pull request by zahiraam**
This patch is joint work with @mdtoguchi.
It adds the `-include-footer` option to pull in the footer.  The syntax is:
`-include-footer=path-to-footer.h` 
This will have the effect of removing the append phase in the compilation process. 
It also implements the required changes to the pre-processed file.

### CODE REVIEW

**Review by maarquitos14 — APPROVED** [review]
LGTM.

File: clang/test/Driver/sycl-cuda-tu-offload.cu [hunk 1]
```diff
@@ -13,43 +13,39 @@
...
+// DEFAULT-PHASES2:             |     +- 11: assembler, {10}, object, (device-cuda, sm_80)
+// DEFAULT-PHASES2:             |  +- 12: offload, "device-cuda (nvptx64-nvidia-cuda:sm_80)" {11}, object
+// DEFAULT-PHASES2:             |  |- 13: offload, "device-cuda (nvptx64-nvidia-cuda:sm_80)" {10}, assembler
+// DEFAULT-PHASES2:             |- 14: linker, {12, 13}, cuda-fatbin, (device-cuda)
+// DEFAULT-PHASES2:          +- 15: offload, "host-cuda-sycl (x86_64-unknown-linux-gnu)" {6}, "device-cuda (nvptx64-nvidia-cuda)" {14}, ir
+// DEFAULT-PHASES2:       +- 16: backend, {15}, assembler, (host-cuda-sycl)
+// DEFAULT-PHASES2:    +- 17: assembler, {16}, object, (host-cuda-sycl)
+// DEFAULT-PHASES2: +- 18: offload, "host-cuda-sycl (x86_64-unknown-linux-gnu)" {17}, object
+// DEFAULT-PHASES2: |                 |- 19: offload, "device-cuda (nvptx64-nvidia-cuda:sm_80)" {9}, ir
+// DEFAULT-PHASES2: |              +- 20: linker, {4, 19}, ir, (device-sycl, sm_80)
```
Comment (mdtoguchi) [inline clang/test/Driver/sycl-cuda-tu-offload.cu:51]: Looks to be an EOF marker problem here.

**Comment by mdtoguchi** [comment]
As I'm the primary author of the driver changes, @hchilama can you review the driver changes?

**Review by hchilama — APPROVED** [review]
LGTM - Driver

**Review by PietroGhg — APPROVED** [review]
Change in Native CPU test lgtm, thank you

**Review by premanandrao — APPROVED** [review]
FE changes LGTM

**Review by maksimsab — APPROVED** [review]
Offloading part LGTM.

**Comment by zahiraam** [comment]
@intel/llvm-gatekeepers Can this PR be merged in? The failure is not related to this patch. Thanks.

**Comment by dm-vodopyanov** [comment]
@sarnex - https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/ContributeToDPCPP.md#commit-message . Please pay attention to that while merging patches, we need such tags at least for creating Release Notes.
FYI, @zahiraam.

**Comment by sarnex** [comment]
@dm-vodopyanov Sorry about that, I will catch it in the future. Thanks for reminding me.

**Pull request merged**